cmd/manager/main.go

```diff
@@ -294,8 +296,9 @@ func main() {
 		BaseCachePath: filepath.Join(cachePath, "unpack"),
 		SourceContextFunc: func(logger logr.Logger) (*types.SystemContext, error) {
 			srcContext := &types.SystemContext{
-				DockerCertPath: caCertDir,
-				OCICertPath:    caCertDir,
+				DockerCertPath:           caCertDir,
+				OCICertPath:              caCertDir,
+				SystemRegistriesConfPath: registriesConfig,
```

Comment:
I suspect this may result in other subtle bugs if there are files in `/etc/containers/registry.conf.d/`, but I don't know for sure. See: https://github.com/containers/image/blob/main/docs/containers-registries.conf.d.5.md

Either way, I think we may need to compute a hash over the entire `/etc/containers` directory in order to guarantee that we catch all relevant file changes. Maybe with `filepath.Walk` that writes every file path and file content to a hasher?

Reply:
+1 on the `.d` config directory

Reply:
Unfortunately `filepath.Walk` does not handle the symlinks that get mounted in the pod, so we'll need to write our own thing unless there's another good library out there somewhere. Agreed here as well though that we should probably just check the entire directory.

From what I understand though @joelanford, K8s updates a symlink in the mounted folder named with a timestamp each time the mounted volume is updated. Could we not just check the filename and see if the timestamp has changed since our last `Unpack`?

Reply:
Oh yeah, if that works, that would be much easier and more efficient.